src/uct/ib/base/ib_md.h

```diff
@@ -353,6 +365,41 @@ typedef ucs_status_t (*uct_ib_md_mem_prefetch_func_t)(uct_ib_md_t *md,
 typedef ucs_status_t (*uct_ib_md_get_atomic_mr_id_func_t)(uct_ib_md_t *md,
                                                           uint8_t *mr_id);
 
+
+/**
+ * Memory domain method to register crossed mkey for memory area.
+ *
+ * @param [in]  ib_md           Memory domain.
+ * @param [in]  flags           UCT memory registration flags.
+ * @param [in]  address         Memory area start address.
+ * @param [in]  length          Memory area length.
+ * @param [out] ib_memh         Memory region handle.
+ *                              Method should initialize lkey & rkey.
+ *
+ * @return UCS_OK on success or error code in case of failure.
+ */
+typedef ucs_status_t (*uct_ib_md_reg_exported_key_func_t)(
+        uct_ib_md_t *ib_md, uint64_t flags, void *address, size_t length,
+        uct_ib_mem_t *ib_memh);
+
+
+/**
+ * Memory domain method to register crossing mkey for memory area.
+ *
+ * @param [in]  ib_md          Memory domain.
+ * @param [in]  flags          UCT memory attach flags.
+ * @param [in]  target_gvmi_id Target GVMI ID.
+ * @param [in]  target_mkey    Target mkey this mkey refers to.
+ * @param [out] ib_memh        Memory region handle.
+ *                             Method should initialize lkey and rkey.
+ *
+ * @return UCS_OK on success or error code in case of failure.
+ */
+typedef ucs_status_t (*uct_ib_md_import_exported_key_func_t)(
```

Comment:
uct_ib_md_import_key_func_t

Reply:
seems unresolved - IMO "import" is better than "import_exported" in several places - since obviously if you import something, it had to be exported..